src/realm/alloc_slab.cpp

```diff
@@ -346,17 +346,24 @@ void SlabAlloc::push_freelist_entry(FreeBlock* entry)
 
 void SlabAlloc::mark_freed(FreeBlock* entry, int size)
 {
+    // For compressed arrays we might have a problem.
```

Comment:
We need to discuss this, capacity does not match allocated space any more for B arrays, it results in some assertions to not be holding.